specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.5-preview.1/backuprestore.json

```diff
@@ -39,6 +39,14 @@
         },
         "description": "Creates a full backup using a user-provided SAS token to an Azure blob storage container.",
         "parameters": [
+          {
+            "name": "storageAuthType",
+            "in": "query",
+            "description": "Type of authentication to use",
+            "required": false,
+            "type": "string",
+            "enum": ["sasToken", "managedIdentity"]
+          },
```

Comment:
Given this is a POST already, why not just add this as a request model property?

Reply:
We wanted to minimize the number of changes to the already existing client code and not break them with the new api version. As a result, after discussing, we thought instead of modifying the body parameters, we thought exposing via optional query parameters helps to allow the newer client to pass the managedIdentity auth mode and existing clients just can continue to use the same request without passing any query parameter even when they upgrade their client.

Reply:
Adding properties in a new API version is not a breaking change. Either way, it's a change for clients. The property - like the parameter - should be optional anyway, so how would it break existing clients?

@johanste or @JeffreyRichter would we want a POST endpoint mixing body and query parameters? We've seen it before as more of a filter, I know, but this is used as more of a directive and seems it should be in the body parameter - whether to use SAS or MSI auth along with the current storage container URI.